Revised the version of several libraries, add @Ignore for several tests

andyglick opened the pull request:
added @Ignore to 4 tests in

CinnamonExpectedConditionsTest.java

bumped selenium-api jar to 3.0.1, was being brought in as a transitive dependency at 2.53.1.

andyglick commented:
the original pull request required a merge from your master, which I didn't realize until after I had posted the 1st PR. This latest version of the PR has passed all of the tests and includes the merge.

davebassan commented:
Hi Andy. Thanks for submitting the PR.

Are you looking to bump selenium support to 3.0.1? I have noticed that you have included a specific dependency on the selenium-api with this PR. Is there a reason for this? My understanding that selenium-java provides transitive dependencies, i.e. selenium-java -> selenium-support -> selenium-api

andyglick commented:
Given that Cinnamon is a Java 8 application and that Selenium 3.0.1 was
released in October it isn't clear to me that it makes sense not to upgrade
the driver to 3.0.1. Here are the release notes:
https://seleniumhq.wordpress.com/category/releases/

If you all aren't interested in upgrading at this time, please let me know
and I can roll back the changes made on behalf of the selenium update and
submit a new pull release.

I've been having fun working in the Cherry Framework projects, so please
keep up the good work.

Andy

On Fri, Jan 27, 2017 at 4:17 PM, Dave Bassan <notifications@github.com>
wrote:

> Hi Andy. Thanks for submitting the PR.
>
> Are you looking to bump selenium support to 3.0.1? I have noticed that you
> have included a specific dependency on the selenium-api with this PR. Is
> there a reason for this? My understanding that selenium-java provides
> transitive dependencies, i.e. selenium-java -> selenium-support ->
> selenium-api
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MagenTys/cinnamon/pull/9#issuecomment-275776959>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAd9hAxW3H9tNlzdLWViCAHlqQhjO4gSks5rWl8EgaJpZM4LwEqT>
> .
>

andyglick commented:
I was hoping that you might have informed me why you refused my pull request. I did spend some time working on it.

If you'd prefer that I stop contributing to your project(s) you have only to ask.

mplord commented:
Apologies if this was not intended - from your comment:

https://github.com/MagenTys/cinnamon/issues/10
'After I posted the pull request I found that there were issues with the code that I submitted. If someone could delete the existing PR, then I'll create a new one when I've got everything cleaned up.'

I thought you still wanted the PR closing - I've re-opened.

andyglick commented:
hello, made a series of changes to the CinnamonExpectedConditionsTest class. I have to admit that I normally think of checking the messages of Exceptions as being a code smell, but since that's what you all were doing I made a simple change. I noticed that the full text string being generated by the code once all of the libraries were advanced was no longer fully deterministic, but the first N characters did prove to provide a reasonable pattern to test against, so I dropped the use of the  ExpectedException class and replaced it with a catch or the expected exception and a String::startsWith method call using the known beginning of the error message text. 

Something odd popped up with the maven build for the webdriver module. For reasons that I don't understand the aspectj-maven-plugin's compile phase started reporting a failed condition and started failing the build. I have reported the error to the aspectj-maven-plugin's issues list at github.

davebassan commented on pom.xml:
```diff
@@ -1,569 +1,724 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Please can you reformat as it is difficult to identify what has changed.

davebassan commented on vendor-support/pom.xml:
```diff
@@ -32,6 +32,13 @@
         <dependency>
             <groupId>org.seleniumhq.selenium</groupId>
             <artifactId>selenium-java</artifactId>
+            <version>${selenium.version}</version>
```
I don't understand what this is doing. It appears to be both including and excluding at the same time.

davebassan commented on webdriver-factory/pom.xml:
```diff
@@ -41,6 +41,63 @@
         </plugins>
     </build>
 
+    <dependencyManagement>
```
This section should reside in the parent pom. Any references to versions of artifacts should be centralised in the parent.

davebassan commented on webdriver-factory/pom.xml:
```diff
@@ -54,7 +111,18 @@
 
         <dependency>
             <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-api</artifactId>
```
Is this needed? I thought that the selenium-api is a transitive dependency of selenium-java.
